File: cp_nim/src/bin/abc240-b-count-distinct-integers.nim
# Created by Ayush Biswas at 2026/07/18 18:15
# https://atcoder.jp/contests/adt_all_20260624_1/tasks/abc240_b
import lib/cpio

# @code begin

import sequtils

var r = newReader()

let
  n = r[int]
  a = r[seq[int]]

echo a.deduplicate(false).len

# @code end

File: cp_rust/src/bin/abc240-b-count-distinct-integers.rs
// Created by Ayush Biswas at 2026/07/07 11:47
// https://atcoder.jp/contests/adt_all_20260624_1/tasks/abc240_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        n: usize,
        a: [usize]
    ) -> usize {
        a.into_iter().unique().count()
    }
}
// @code end
